### ISSUE TITLE
[BUG] pmdarima 2.0.0 incompatibility - return indices

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#3300]
Soft dependency `pmdarima` 2.0.0 was released a few hours ago, and it seems to break our `ARIMA` interface.

Suspicion: return indices are now absolute and no longer relative to the cutoff.

**Comment by ciberger** [comment]
This breaks function

**Comment by fkiraly** [comment]
@ciberger, @ngupta23, this relates to an old discussion here:
https://github.com/alan-turing-institute/sktime/issues/1480

There are two opposing poles in the discussion:
* we should not have defensive version bounds, since that is not nice for reverse dependencies and, if everyone does this, it might ultimately lead to unsatisfiable constraint combinations
* we should have defensive version bounds and bump them every release, because it can break production code that does not itself have defensive version bounds

I'm not sure where the discussion was had since the issue #1480 only has my posts, but I remember a notably heated discussion  (@aiwalter, @lmmentel, @freddyaboulton, where is all th discussion?), and the current majority opinion among core developers is "we should not have defensive version bounds and should gradually remove the ones in place"

Perhaps we can find a compromise and put defensive version founds for MAJOR but not MINOR releases?

**Comment by fkiraly** [comment]
Either way, the 0.13.2 release will have the pmdarima bound in.

**Comment by aiwalter** [comment]
can someone paste the exception message here? @ciberger what is "data" in your case?

**Comment by aiwalter** [comment]
i am getting 
```python
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
File c:\Users\mf-wa\Desktop\sktime\pmdarima\pmdarima\__check_build\__init__.py:44, in <module>
     43 try:
---> 44     from ._check_build import check_build
     45 except ImportError as ie:

ModuleNotFoundError: No module named 'pmdarima.__check_build._check_build'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
Untitled-1.ipynb Zelle 1 in <cell line: 2>()
      [1](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#ch0000000untitled?line=0) from sktime.datasets import load_airline
----> [2](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#ch0000000untitled?line=1) from sktime.forecasting.arima import AutoARIMA
      [3](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#ch0000000untitled?line=2) y = load_airline()
      [4](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#ch0000000untitled?line=3) forecaster = AutoARIMA(sp=12, d=0, max_p=2, max_q=2, suppress_warnings=True)  

File c:\Users\mf-wa\miniconda3\envs\sktime39\lib\site-packages\sktime\forecasting\arima.py:12, in <module>
      9 from sktime.forecasting.base.adapters._pmdarima import _PmdArimaAdapter
     10 from sktime.utils.validation._dependencies import _check_soft_dependencies
---> 12 _check_soft_dependencies("pmdarima", severity="warning")
     15 class AutoARIMA(_PmdArimaAdapter):
     16     """Wrapper of the pmdarima implementation of fitting Auto-(S)ARIMA(X) models.
     17
```
but this seems unrelated to `pmdarima` version?

**Comment by ciberger** [comment]
I suspect the issue appears on the `fh` parameter from `predict`. Defined as list or with `ForecastingHorizon` triggers the error.

Here you have the error message. 
```python
/local_disk0/.ephemeral_nfs/envs/pythonEnv-958c1258-3728-45b7-bb5a-105442b141b8/lib/python3.8/site-packages/sktime/forecasting/base/_fh.py:100: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
  values = pd.Int64Index(values, dtype=int)
Out[27]: ForecastingHorizon([1], dtype='int64', is_relative=True)
```

I highly thought this is generated due an index issue in the input data.

**Comment by aiwalter** [comment]
I can just see a warning, please share full trace

**Comment by ciberger** [comment]
Full trace
```python
KeyError                                  Traceback (most recent call last)
<command-682151211076022> in <module>
      3   forecaster = AutoARIMA(sp=7, d=1, max_p=2, max_q=2, suppress_warnings=True)
      4   forecaster.fit(data)
----> 5   forecast = forecaster.predict(fh=[1])

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/sktime/forecasting/base/_base.py in predict(self, fh, X)
    323         # we call the ordinary _predict if no looping/vectorization needed
    324         if not self._is_vectorized:
--> 325             y_pred = self._predict(fh=fh, X=X_inner)
    326         else:
    327             # otherwise we call the vectorized version of predict

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/sktime/forecasting/base/adapters/_pmdarima.py in _predict(self, fh, X)
     94         # all values are out-of-sample
     95         if fh.is_all_out_of_sample(self.cutoff):
---> 96             return self._predict_fixed_cutoff(fh_oos, X=X)
     97 
     98         # all values are in-sample

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/sktime/forecasting/base/adapters/_pmdarima.py in _predict_fixed_cutoff(self, fh, X, return_pred_int, alpha)
    221             return result[0], pred_ints
    222         else:
--> 223             return pd.Series(result[fh_idx], index=fh_abs)
    224 
    225     [0;32mdef _predict_interval(self, fh, X=None, coverage=0.90):

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/series.py in __getitem__(self, key)
    982             return self._get_values(key)
    983 
--> 984         return self._get_with(key)
    985 
    986     def _get_with(self, key):

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/series.py in _get_with(self, key)
   1017             #  (i.e. self.iloc) or label-based (i.e. self.loc)
   1018             if not self.index._should_fallback_to_positional:
-> 1019                 return self.loc[key]
   1020             else:
   1021                 return self.iloc[key]

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/indexing.py in __getitem__(self, key)
    965 
    966             maybe_callable = com.apply_if_callable(key, self.obj)
--> 967             return self._getitem_axis(maybe_callable, axis=axis)
    968 
    969     def _is_scalar_access(self, key: tuple):

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/indexing.py in _getitem_axis(self, key, axis)
   1189                     raise ValueError("Cannot index with multidimensional key")
   1190 
-> 1191                 return self._getitem_iterable(key, axis=axis)
   1192 
   1193             # nested tuple slicing

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/indexing.py in _getitem_iterable(self, key, axis)
   1130 
   1131         # A collection of keys
-> 1132         keyarr, indexer = self._get_listlike_indexer(key, axis)
   1133         return self.obj._reindex_with_indexers(
   1134             {axis: [keyarr, indexer]}, copy=True, allow_dups=True

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/indexing.py in _get_listlike_indexer(self, key, axis)
   1325         axis_name = self.obj._get_axis_name(axis)
   1326 
-> 1327         keyarr, indexer = ax._get_indexer_strict(key, axis_name)
   1328 
   1329         return keyarr, indexer

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/indexes/base.py in _get_indexer_strict(self, key, axis_name)
   5780             keyarr, indexer, new_indexer = self._reindex_non_unique(keyarr)
   5781 
-> 5782         self._raise_if_missing(keyarr, indexer, axis_name)
   5783 
   5784         keyarr = self.take(indexer)

/local_disk0/.ephemeral_nfs/envs/pythonEnv-137bba7a-cea1-46dc-b8d2-db0341422723/lib/python3.8/site-packages/pandas/core/indexes/base.py in _raise_if_missing(self, key, indexer, axis_name)
   5840                 if use_interval_msg:
   5841                     key = list(key)
-> 5842                 raise KeyError(f"None of [{key}] are in the [{axis_name}]")
   5843 
   5844             not_found = list(ensure_index(key)[missing_mask.nonzero()[0]].unique())

KeyError: "None of [Int64Index([0], dtype='int64')] are in the [index]"
```

**Comment by aiwalter** [comment]
@ciberger are you getting any deprecation warning on the last old version that is working if you set `suppress_warnings=False`? If not, we should complain at `pmdarima` about it. In future, we can avoid such issues if we have a CI/CD pipeline step that checks for deprecation warnings so we can see them before the actual deprecation is happening

**Comment by fkiraly** [comment]
I don't think there was any deprecation warning coming from `pmdarima`, afaik this happened without a forewarning.

The error above is precisely the one that CI/CD also caught already when I opened this issue.

The root cause I do not fully understand, since index behaviour seems to be one way or the other (new/old), inconsistently.

**Comment by fkiraly** [comment]
@ciberger, 0.13.2 is released now with a <2.0.0 bound on `pmdarima`, which fixes the symptom temporarily. Any insight on what is going on with `pmdarima 2.0.0` would be appreciated.
